documentation/projects/proposals/additional_search_views/20230719-implementation_plan_additional_search_views.md

```diff
@@ -111,35 +121,194 @@ for fuzzy matches.
 **Finally**, we apply the ranking and sorting parameters, and "highlight" the
 fields that were matched in the results.
 
-The new search controller should allow for using different filters for the first
-step and to not use the full-text search. We should also create a new serializer
-for collection search requests. It should include the common parameters for
-`list` requests, such as `page` and `page_size`, and the parameters for the
-exact matches: `tag`, `creator` and `source`.
+The search controller needs to be updated to extract the first 2 steps into a
+`build_search_query` method. A new `build_collection_query` should be added and
+used when the `collection` parameter is present. This method should create a
+filter query for the relevant field and value.
+
+The pagination and dead link cleanup should be the same for additional search
+queries as for the default search queries.
+
+#### Search request serializer updates
+
+_The initial version of this plan proposed to add new API endpoints and use path
+parameters for the values (e.g., `/image/tag/cat`) to make the URLs more
+readable, easier to share, will be easier to cache or perform cache invalidation
+required by [#1969](https://github.com/WordPress/openverse/issues/1969).
+However, since tags and creator names contain characters that are special for
+path segments (`/`, `?` and `&`) that cannot be properly encoded, this approach
+turned out to be not feasible._
+
+```{note}
+The new query parameters (`collection` and `tag`) will have `unstable__` prefix until this project is
+launched. In the text below, for brevity, the prefix is omitted.
+```
+
+The collections will use the `search` endpoint with `collection` query parameter
+set to `tag`, `creator` or `source`, and the following additional query
+parameters:
+
+- `collection=tag` will require `tag` parameter to be set
+- `collection=source` will require `source` parameter to be set
+- `collection=creator` will require both `source` and `creator` parameter to be
+  set.
```

Comment:
This is the part that gets confusing for me because, in the "Expected Outcomes" section, the new URLs are shown as belonging to a new `collection/` endpoint and not a tag with a value like the ones shown here. What would be the final form, as `/<mediaType>/collection?tag=tagName` or `/<mediaType>/?collection=tag&tag=<tag_name>` or maybe like `/<mediaType>/?collection=tag&tagName=<tag_name>`? 

This is fundamental because if they end up in the form of `/<mediaType>/collection?tag=<tag_name>`, then there is not much difference from the original proposal of creating new endpoints, which my experience tells me it should be the easiest and cleanest path, at least for **tags** and **sources**. The `creator` makes perfect sense as a parameter for the `source` collection as it's currently portrayed.

Reply:
I added sample URLs for API in the expected outcomes and made the section easier to scan.